storage/src/journal/mod.rs

```diff
@@ -37,4 +37,6 @@ pub enum Error {
     InvalidItem(u64),
     #[error("invalid rewind: {0}")]
     InvalidRewind(u64),
+    #[error("invalid index: {0}")]
+    InvalidIndex(u64),
```

Comment:
I wouldn't expose the fact that we index blobs using integers to the external API like this.